### PR TITLE
[#6] update getLegislation to use Postgres

### PR DESCRIPTION
**Pull request by mcqueen1987**
Update getLegislation.js to use similar code to getCases

### CODE REVIEW

**Comment by mcqueen1987** [comment]
Hi @williamparry could you please help to review the code? 
besides, am not sure about the function insertLegislation(), should I insert all data at once or insert one by one? thanks

File: getLegislation.js [hunk 1]
```diff
@@ -1,49 +1,65 @@
...
-	console.log(`Inserting ${insertValues.length} legislation`);
-
-	if (insertValues.length > 0) {
-		await pipeline_connection.query(
-			"INSERT INTO legislation (title, link, year, alerts) VALUES ?",
-			[insertValues]
-		);
-		pipeline_connection.end();
-	}
+    try {
```
Comment (williamparry) [inline getLegislation.js]: Can you please wrap this in a transaction and rollback if it fails, or put it in one statement.

File: common/setup.js [hunk 2]
```diff
@@ -4,52 +4,53 @@ const uuidv1 = require('uuid/v1');
...
+    const sessionId = resumeSessionId || uuidv1();
+    const cacheDir = path.join(rootDir, '.cache', sessionId);
+    const logDir = path.join(rootDir, '.logs', sessionId);
+
+    if (!env) {
+        throw new Error('Missing env');
+    }
+
+    require('dotenv').config({
+        path: rootDir + '/' + env
```
Comment (williamparry) [inline common/setup.js]: Follow the convention of the other code in the repo e.g. `.env.local`

File: common/setup.js [hunk 3]
```diff
@@ -4,52 +4,53 @@ const uuidv1 = require('uuid/v1');
...
-
-	return {
-		sessionId,
-		cacheDir,
-		logDir,
-		pgPromise,
-		connection
-	};
+    const options = {
+        capSQL: true, // capitalize all generated SQL
```
Comment (williamparry) [inline common/setup.js]: Why is this necessary?

File: getDataFile/pcoLegislation.js [hunk 4]
```diff
@@ -1,35 +1,34 @@
...
 
 if (require.main === module) {
-	try {
-		throw new Error("Cannot be run individually.")
-		//run();
-	} catch (ex) {
-		console.log(ex);
-	}
+    try {
+        run();
```
Comment (williamparry) [inline getDataFile/pcoLegislation.js]: Is there a reason this was uncommented?

File: common/setup.js (hunk 3 above)
Comment (mcqueen1987) [reply common/setup.js]: 'capSQL: true' makes SQL tokens capitalized which makes SQL easy to read. change from 'select * from' to 'SELECT * FROM'. It’s a matter of preference and I think it's better be capitalized.

File: common/setup.js (hunk 2 above)
Comment (mcqueen1987) [reply common/setup.js]: Yes, you are right, if I use file .env.local but not .env, then I don't need to modify the code.

File: getDataFile/pcoLegislation.js (hunk 4 above)
Comment (mcqueen1987) [reply getDataFile/pcoLegislation.js]: I thought:
1, legislation is an independent module that could run independently and it's tables are independent too.
2, if 1 is correct, we just need to run command line 'getLegislation.js --env=local --datasource=pco' regularly to get all legislation data.
3,  'throw new Error("Cannot be run individually.")' ---- this is just a debug info right? or  I have missed something?

if the info above is not correct, leave a msg or give me a call when you free. thanks :)

File: getLegislation.js (hunk 1 above)
Comment (mcqueen1987) [reply getLegislation.js]: good idea, I gonna fix it later, thanks :)
